packages/rum/src/userActionCollection.ts

```diff
@@ -0,0 +1,200 @@
+//
+//                               (Start)
+//                   .--------------'----------------------.
+//                   v                                     v
+//     [Wait for a page activity ]          [Wait for a maximum duration]
+//     [timeout: VALIDATION_DELAY]          [  timeout: MAX_DURATION    ]
+//          /                  \                           |
+//         v                   v                           |
+//  [No page activity]   [Page activity]                   |
+//         |                   |,----------------------.   |
+//         v                   v                       |   |
+//     (Discard)     [Wait for a page activity]        |   |
+//                   [   timeout: END_DELAY   ]        |   |
+//                       /                \            |   |
+//                      v                 v            |   |
+//             [No page activity]    [Page activity]   |   |
+//                      |                 |            |   |
+//                      |                 '------------'   |
+//                      '-----------. ,--------------------'
+//                                   v
+//                                 (End)
+//
+// Note: because MAX_DURATION > VALIDATION_DELAY, we are sure that if the user action is still alive
+// after MAX_DURATION, it has been validated.
+
+// Delay to wait for a page activity to validate the user action
+const USER_ACTION_VALIDATION_DELAY = 100
+// Delay to wait after a page activity to end the user action
+const USER_ACTION_END_DELAY = 100
+// Maximum duration of a user action
+export const USER_ACTION_MAX_DURATION = 10_000
+
+export function startUserActionCollection(lifeCycle: LifeCycle) {
+  function processClick(event: Event) {
+    if (!(event.target instanceof Element)) {
+      return
+    }
+
+    const content = getElementContent(event.target)
+
+    const { observable: pageActivitiesObservable, stop: stopPageActivitiesTracking } = trackPagePageActivities(
+      lifeCycle
+    )
+
+    newUserAction(pageActivitiesObservable, (userActionDetails) => {
+      stopPageActivitiesTracking()
+      if (userActionDetails) {
+        lifeCycle.notify(LifeCycleEventType.USER_ACTION_COLLECTED, {
+          duration: userActionDetails.duration,
+          id: userActionDetails.id,
+          name: content,
+          startTime: userActionDetails.startTime,
+          type: UserActionType.CLICK,
+        })
+      }
+    })
+  }
+
+  addEventListener('click', processClick, { capture: true })
+
+  return {
+    stop() {
+      removeEventListener('click', processClick, { capture: true })
+    },
+  }
+}
+
+let currentUserAction: { id: string; startTime: number } | undefined
+
+export function getUserActionId(time: number): string | undefined {
+  if (currentUserAction && time >= currentUserAction.startTime) {
+    return currentUserAction.id
+  }
+  return undefined
+}
+
+interface UserActionDetails {
```

Comment:
What about `UserActionCompleteEvent` in order to be more consistent with other namings?
then, we we could rename `createDetails` into something like `completeUserAction`.